### DISCUSSION TITLE
General Discussion about netscanner

### DISCUSSION THREAD

**Discussion by Chleba** [Chleba/netscanner#25] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zoza1982** [comment]
Hi there, thank you for the amazing tool that you built for the community. I am a big fan. I wanted to contribute , being AI augmented Infrastructure engineer , I use a lot of Agentic tooling and I am able to make a large contribution quite quickly.  I did not wanted to overwhelm you with a PR just yet until you review my fork and test it yourself. Please feel free to review in my fork all  closed PR's and commits.

My fork: https://github.com/zoza1982/netscanner 

Let me know , I would love to open PR against your repo as its original repo as I don't have intention to maintain a fork.
And I am able to contribute directly into your repo in the future. We can discuss further ideas.
I am happy to connect  zoran.vukmirica@gmaill.com / https://www.linkedin.com/in/zoran-vukmirica-22275a17/

**Reply by Chleba** [reply]
Hi @zoza1982,

I have little bit of time for your changes. I will test your fork, take a longer look at the generated code and let you know if I could use that code and how do we merge it (if in one PR - squashed or more).
Thanks again for your patience and your hard work.

**Reply by zoza1982** [reply]
Sounds good! thanks!

**Reply by Chleba** [reply]
ok, I did checked Your changes, checked out your fork and try it and there is a few errors.
- when I scan all clients and trying to scan ports at multiple clients I get an error - out of variants or something (idk what is that mean and i didn't put it there so it's in your changes)
- i would really like to remove all of that endless comments that those agents put in everywhere
- i don't have any IPv6 network to test Your added feature to scan IPv6 networks, so my question is: "did you test that  thoroughly ?"

**Reply by zoza1982** [reply]
Sorry I've been traveling. 
I will get these issue fixed. 
I had limited ipv6 testing too but it I did run a test against google ipv6 DNS. I will share more details

**Reply by zoza1982** [reply]
Thanks for finding that issue when scannig multiple hosts. It has ben fixed and merged into my main branch. I was able to reproduce same issue and verified after the fix. It works now.

Excessive comments are removed. I agree, it was a bit too much.
Feel free to refresh my main branch and re-test it.

I re-checked the Ipv6 testing, and here is the instructions :
-----------------------------------------------------------
 Quick IPv6 Testing

  1. Test with Google DNS (Easiest)

  sudo ./target/release/netscanner

  In Discovery tab:
  - Enter CIDR: 2001:4860:4860::8888/128
  - Press s to scan
  - Should discover Google's IPv6 DNS server

  ---
  2. Test with Cloudflare DNS

  In Discovery tab:
  - Enter: 2606:4700:4700::1111/128
  - Press s to scan

**Reply by zoza1982** [reply]
- NDP (Neighbor Discovery) only works for local network hosts
  - Internet hosts (Google DNS, etc.) won't show MAC addresses
  - This is expected and normal

**Comment by zoza1982** [comment]
@Chleba is everything OK with the PR? Are you planning to merge it? If not, I can withdraw if something is not good.
